conductor/lib/pyside_utils.py

```diff
@@ -217,6 +217,76 @@ def launch_yes_no_dialog(title, message, show_not_again_checkbox=True, parent=No
     dont_notify_again = dialog.checkBox.isChecked()
     return bool(yes), dont_notify_again
 
+def launch_yes_no_cancel_dialog(title, message, show_not_again_checkbox=True, parent=None):
+    '''
+    Launch a dialog box that has "yes", "no" and "cancel" buttons.
+    Optionally display a checkbox that asks whether to show this dialog box again.
+    This is different from the Yes/No style of dialog since it offers three choices.
+    As a consequence, the return code needs to return one of three statuses.
+    return the result of this dialog (0/1/2) for 'Yes'/'No'/'Cancel' respectively, and
+    whether the user checked the "Don't ask again" checkbox (True/False).
+    '''
+
+    dialog = QtWidgets.QDialog(parent=parent)
+    dialog.setWindowTitle(title)
+    dialog.verticalLayout = QtWidgets.QVBoxLayout(dialog)
+
+    # Create the dialogs main message (Qlabel)
+    dialog.label = QtWidgets.QLabel(dialog)
+    dialog.label.setAlignment(QtCore.Qt.AlignCenter)
+    dialog.label.setTextInteractionFlags(dialog.label.textInteractionFlags() | QtCore.Qt.TextBrowserInteraction)
+    dialog.label.setTextFormat(QtCore.Qt.RichText)
+    dialog.label.setOpenExternalLinks(True)
+    dialog.label.setText(message)
+    dialog.verticalLayout.addWidget(dialog.label)
+
+    dialog.widget = QtWidgets.QWidget(dialog)
+    dialog.horizontalLayout = QtWidgets.QHBoxLayout(dialog.widget)
+    dialog.horizontalLayout.setContentsMargins(-1, -1, -1, 0)
+    dialog.horizontalLayout.setObjectName("horizontalLayout")

```

Comment:
Is there a need to explicitly set the layout name here?

Reply:
I used the layout from the other one, my assumption was that it was needed.

Reply:
removed this line